apps/src/lib/tools/jsdebugger/MercatorMap.jsx

```diff
@@ -0,0 +1,97 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {geoMercator, geoPath} from 'd3-geo';
+import {feature} from 'topojson-client';
+
+const PROJECTION_RADIUS = 70;
+const MAP_WIDTH = 400;
+const MAP_HEIGHT = 400;
+
+class MercatorMap extends React.Component {
+  static propTypes = {
+    data: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
+  };
+
+  state = {
+    mercator: '',
+    data: this.props.data
+  };
+
+  async componentDidMount() {
+    let value = await this.mercator();
+    this.setState({
+      mercator: value
+    });
+  }
+
+  projection = geoMercator()
+    .scale(PROJECTION_RADIUS)
+    .translate([MAP_WIDTH / 2, MAP_HEIGHT / 2]);
+
+  displayLocation() {
+    if (this.state.data.constructor === Array) {
+      return this.state.data.map((location, i) => {
+        return (
+          <circle
+            fill={'red'}
+            r={3}
+            transform={`translate(${this.projection([
+              location.long,
+              location.lat
+            ])})`}
+          />
+        );
+      });
+    } else {
+      return (
+        <circle
+          fill={'red'}
+          r={5}
+          transform={`translate(${this.projection([
+            this.state.data.long,
+            this.state.data.lat
+          ])})`}
+        />
+      );
+    }
+  }
+
+  async mercator() {
+    let displayWorldMap = async function() {
+      const worldMapUrl = 'https://unpkg.com/world-atlas@1/world/110m.json';
```

Comment:
Might need to be a longer discussion with @ryansloan or @joshlory (or someone who is better at reading licenses than me), but do you know if we are clear to use this data according to its license? https://unpkg.com/world-atlas@1.1.4/LICENSE.md

Reply:
Ok, I'll ping both to ask about this!